### PR TITLE
Improve links to example workflows

### PR DESCRIPTION
**Pull request by emcfarlane**
This PR adds a new section for linking to the push on changes example, which was a requested example. I've also re-ordered the README.md to move the `Customizing Behaviour` section below the `Version` and `Authentication` sections. I think this reads better for the user, as the user will generally want to customize the `Version` and `Authentication` before customizing the defaults, which is a more advanced use case.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -229,6 +169,88 @@ Alternatively, you may wish to pre-checkout the base branch for breaking changes
...
+    # Caution: This workflow could miss changes if the paths are not correctly specified.
+    - '**.proto'
+    - '**/buf.yaml'
+    - '**/buf.lock'
+    - '**/buf.md'
+    - '**/README.md'
+    - '**/LICENSE'
+```
+
+See the [push-on-changes.yaml](examples/push-on-changes.yaml) example.
```
Comment (emcfarlane) [inline README.md:251]: This is new.

File: README.md [hunk 2]
```diff
@@ -229,6 +169,88 @@ Alternatively, you may wish to pre-checkout the base branch for breaking changes
...
+#### Only push on changes
+
+To push only on changes to the protos, restrict the push step for any changes to buf releated files.
+This can be achieved by using the `paths` filter on the `push` event.
+
+```yaml
+push:
+  paths:
+    # Caution: This workflow could miss changes if the paths are not correctly specified.
+    - '**.proto'
```
Comment (nicksnyder) [inline README.md:243]: Does this need to be `**/*.proto`?

Comment (emcfarlane) [reply README.md:243]: I don't know where the docs specifically for the glob language are but shown in this [example](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-the-head-or-base-branch-of-a-pull-request) and from testing only `**.proto` is needed to trigger on any proto file.

Comment (emcfarlane) [reply README.md:243]: '**/README.md' will trigger for root and any subdirs.

**Review by nicksnyder — APPROVED** [review]

**Pull request merged**